### PR TITLE
Make UUID function a standard function

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/type/UuidOperators.java [hunk 1]
```diff
@@ -44,11 +46,21 @@
...
+import static java.util.UUID.randomUUID;
 
 public final class UuidOperators
 {
     private UuidOperators() {}
 
+    @Description("generates a random UUID")
+    @ScalarFunction(deterministic = false)
+    @SqlType(StandardTypes.UUID)
+    public static Slice uuid()
```
Comment (findepi) [inline presto-main/src/main/java/io/prestosql/type/UuidOperators.java]: You could add this in `UuidFunctions` (a new class). No strong opinion.

File: presto-main/src/test/java/io/prestosql/type/TestUuidOperators.java [hunk 2]
```diff
@@ -33,6 +33,12 @@
 public class TestUuidOperators
         extends AbstractTestFunctions
 {
+    @Test
+    public void testRandomUuid()
+    {
+        tryEvaluateWithAll("uuid()", UUID);
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/type/TestUuidOperators.java]: You could verify the value well-formedness.
eg it should be not null, and for test purposes we can assume it should not be all zeros.

**Comment by martint** [comment]
We need to make sure the function is classified as non-deterministic. Otherwise, it might lead to incorrect results due to constant folding

File: presto-main/src/main/java/io/prestosql/type/UuidOperators.java (hunk 1 above)
Comment (findepi) [reply presto-main/src/main/java/io/prestosql/type/UuidOperators.java]: nvrm

**Comment by findepi** [comment]
@martint right. I forgot that `@ScalarFunction(deterministic = false)` is still not sufficient.
@electrum see `io.prestosql.sql.planner.DeterminismEvaluator.Visitor#visitFunctionCall`

File: presto-main/src/main/java/io/prestosql/type/UuidOperators.java (hunk 1 above)
Comment (electrum) [reply presto-main/src/main/java/io/prestosql/type/UuidOperators.java]: I briefly considered that but decided it was overkill for one function and we don't expect to have more anytime soon.

File: presto-main/src/test/java/io/prestosql/type/TestUuidOperators.java (hunk 2 above)
Comment (electrum) [reply presto-main/src/test/java/io/prestosql/type/TestUuidOperators.java]: The non-null part is handled by the expected type assertion. I tried to add a test to validate they are unique, but the non-determinism makes it tricky due to the way the function asserts work.

**Comment by electrum** [comment]
I fixed the determinism evaluator.

File: presto-main/src/test/java/io/prestosql/type/TestUuidOperators.java (hunk 2 above)
Comment (findepi) [reply presto-main/src/test/java/io/prestosql/type/TestUuidOperators.java]: is there a ATQ's `computeActual` equivalent?

**Pull request merged**